src/components/Redux/RocketsSlice/RocketSlice.js

```diff
@@ -1,13 +1,37 @@
-import { createSlice } from '@reduxjs/toolkit';
-import data from '../../TempRocketData';
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
+/* import data from '../../TempRocketData'; */
+
+const url = 'https://api.spacexdata.com/v3/rockets';
 
 const initialState = {
-  RocketDatile: data,
+  RocketDatile: [],
+  loading: false,
+  error: null,
 };
 
+export const rocketItems = createAsyncThunk('rocket', async () => {
+  const response = await fetch(url);
+  const result = response.json();
+  return result;
+});
+
 const rocketSlice = createSlice({
   name: 'rocket',
   initialState,
+  extraReducers: {
+    [rocketItems.pending]: (state) => {
```

Comment:
When using extraReducers is needed to call the (builder) parameter to define and add "extra" reducers to the slice of the store created. Check missionsSlice on this [PR](https://github.com/evillaz/react-redux-group-project/pull/29/commits/5e3eba2e6ff484f9113dffa1277c25fcebfc2150#)